src/main/java/appeng/services/export/MinecraftItemCSVExporter.java

```diff
@@ -0,0 +1,261 @@
+		private final ExportMode mode;
+
+		public TypeExtractFunction( @Nonnull final String itemName, @Nonnull final ExportMode mode )
+		{
+			this.itemName = Preconditions.checkNotNull( itemName );
+			Preconditions.checkArgument( !itemName.isEmpty() );
+
+			this.mode = Preconditions.checkNotNull( mode );
+		}
+
+		@Nullable
+		@Override
+		public String apply( @Nullable ItemStack input )
+		{
+			if( input == null )
+			{
+				AELog.debug( "null type extracted" );
+
+				return null;
+			}
+			else
+			{
+				AELog.debug( "Type extracting %s", input.getDisplayName() + ":" + input.getItemDamage() );
+			}
+
+			final List<String> joinedBlockAttributes = Lists.newArrayListWithCapacity( 5 );
+			final int meta = input.getItemDamage();
+			final String metaName = this.itemName + ':' + meta;
+			final String localization = input.getDisplayName();
+
+			joinedBlockAttributes.add( metaName );
+			joinedBlockAttributes.add( localization );
+
+			if( this.mode == ExportMode.VERBOSE )
+			{
+				final Item item = input.getItem();
+				final String unlocalizedItem = input.getUnlocalizedName();
+				final Block block = Block.getBlockFromItem( item );
+				final boolean isBlock = !block.equals( Blocks.air );
+				final String className = input.getClass().getName();
+
+				joinedBlockAttributes.add( unlocalizedItem );
+				joinedBlockAttributes.add( Boolean.toString( isBlock ) );
+				joinedBlockAttributes.add( className );
+			}
+
+			return Joiner.on( ", " ).skipNulls().join( joinedBlockAttributes );
+		}
+	}
+
+
+	/**
+	 * transforms an item into a row representation of the CSV file
+	 */
+	private static final class ItemRowExtractFunction implements Function<Item, String>
+	{
+		@Nonnull
+		private final FMLControlledNamespacedRegistry<Item> itemRegistry;
+		@Nonnull
+		private final ExportMode mode;
+
+		/**
+		 * @param itemRegistry used to retrieve the name of the item
+		 * @param mode extracts more or less information from item depending on mode
+		 */
+		ItemRowExtractFunction( @Nonnull final FMLControlledNamespacedRegistry<Item> itemRegistry, ExportMode mode )
+		{
+			this.itemRegistry = Preconditions.checkNotNull( itemRegistry );
+			this.mode = Preconditions.checkNotNull( mode );
+		}
+
+		@Nullable
+		@Override
+		public String apply( @Nullable Item input )
+		{
+			if( input == null )
+			{
+				AELog.debug( "Exporting nothing" );
+
+				return null;
+			}
+			else
+			{
+				AELog.debug( "Exporting input %s with subtypes: %b", input.getUnlocalizedName(), input.getHasSubtypes() );
+			}
+
+			final String itemName = this.itemRegistry.getNameForObject( input );
+			final boolean hasSubtypes = input.getHasSubtypes();
+			if( hasSubtypes )
+			{
+				final CreativeTabs creativeTab = input.getCreativeTab();
+				final List<ItemStack> stacks = Lists.newArrayList();
+
+				// modifies the stacks list and adds the different sub types to it
+				try
+				{
+					input.getSubItems( input, creativeTab, stacks );
+				}
+				catch( Exception ignored )
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)